src/color.c

```diff
@@ -1,28 +1,26 @@
 #include "proto.h"
 
+#define formatpair(from, to) (((from) << 2) & (PLUM_COLOR_MASK << 2) | (to) & PLUM_COLOR_MASK)
+
 void plum_convert_colors (void * restrict destination, const void * restrict source, size_t count, unsigned long to, unsigned long from) {
   if (!(source && destination && count)) return;
-  if ((from & (PLUM_COLOR_MASK | PLUM_ALPHA_INVERT)) == (to & (PLUM_COLOR_MASK | PLUM_ALPHA_INVERT))) {
-    memcpy(destination, source, plum_color_buffer_size(count, to));
-    return;
-  }
-  #define convert(sp) do                                                                                       \
-    if ((to & PLUM_COLOR_MASK) == PLUM_COLOR_16)                                                               \
-      for (uint16_t * dp = destination; count; count --) *(dp ++) = plum_convert_color(*(sp ++), from, to);    \
-    else if ((to & PLUM_COLOR_MASK) == PLUM_COLOR_64)                                                          \
-      for (uint64_t * dp = destination; count; count --) *(dp ++) = plum_convert_color(*(sp ++), from, to);    \
-    else                                                                                                       \
-      for (uint32_t * dp = destination; count; count --) *(dp ++) = plum_convert_color(*(sp ++), from, to);    \
-  while (false)
-  if ((from & PLUM_COLOR_MASK) == PLUM_COLOR_16) {
-    const uint16_t * sp = source;
-    convert(sp);
-  } else if ((from & PLUM_COLOR_MASK) == PLUM_COLOR_64) {
-    const uint64_t * sp = source;
-    convert(sp);
-  } else {
-    const uint32_t * sp = source;
-    convert(sp);
+  #define convert(frombits, tobits) do {                                   \
```

Comment:
You already had the `formatpair` macro below, may as well use it here and save a couple of lines + make the to-from paired logic clearer.